java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/MultiModuleNodeFactory.java

```diff
@@ -557,12 +561,40 @@ public String getHtmlDisplayName() {
                         null,
                         SystemAction.get(FileSystemAction.class ),
                         null,
-                        SystemAction.get(ToolsAction.class )
+                        SystemAction.get(ToolsAction.class ),
+                        //Added new action to delete module
+                        new DeleteAction("Delete") // NOI18N
                     };
                 }
                 return actions;
             }
         }
+        
+        private class DeleteAction extends AbstractAction {
```

Comment:
Please don't reinvent the wheel. The explorer view already handles the confirmation - override `canDestroy` and `destroy`  in the `Node` implementation and you should be done.